src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/NuGetDependenciesSubTreeProvider.cs

```diff
@@ -544,32 +588,33 @@ public override Task<IEnumerable<IDependencyNode>> SearchAsync(IDependencyNode n
             }
         }
 
-        private IList<IDependencyNode> SearchRecursive(DependencyMetadata rootDependency,
-                                                       HashSet<DependencyMetadata> flatMatchingDependencies)
+        public void UpdateNodeChildren(string packageItemSpec, IDependencyNode originalNode)
```

Comment:
After reading the code a few times I understand what's happening but can you please add some comments about the interface and the contract etc.